Jenkinsfile

```diff
@@ -124,30 +117,60 @@ def getTests(String baseUrl) {
 
 def getScenarios(String baseUrl, String testId){
     def scenarios = [[Id: "-1", Name: "Default"]]
-    def isScenarioChoosen = "${params['Run with scenarios']}".toBoolean()
+    def isScenarioChoosen = "${params['Run with datasets']}".toBoolean()
     if (isScenarioChoosen) {
-        def jsonSlurper = new JsonSlurper()
         URL url = new URL("${baseUrl}/projects/${params['Project']}/tests/${testId}/scenarios")
-        HttpURLConnection conn = url.openConnection()
-        conn.setRequestMethod("GET")
-        conn.setDoInput(true)
-        conn.setRequestProperty("Authorization", "APIKEY $useMangoApiKey")
-        conn.connect()
-        String content
-        def responseCode = conn.responseCode
-        if (responseCode == 200) {
-            InputStream inputStream = conn.getInputStream()
-            content = inputStream.getText()
-        } else {
-            throw new Exception("Scenarios get request failed with code: ${responseCode}")
-        }
-        def scenarioPage = jsonSlurper.parseText(content)
+        String content = getConnectionContent(url, "Scenarios")
+        def scenarioPage = new JsonSlurper().parseText(content)
         if (scenarioPage != null) {
             scenarioPage.each { scenario ->
-                echo "${scenario.Name}"        
+                echo "${scenario.Name}"
                 scenarios << [Id: scenario.Id, Name: scenario.Name]
             }
         }
     }
     return scenarios
+}
+
+def getEnvId(String baseUrl) {
+    String cursor = ""
+    echo "Retrieving environment based on EnvName"
+    while(true) {
```

Comment:
Let's establish the context first. This method has two responsibilities
1. return environment id
2. throw NoSuchElementException

Now, let's look at the loop. Other than above outcomes, can it produce cause more outcomes?
1. while true -> infinite loop 
2. An unhandled error will be thrown if url is unavailable or api response is not correct or has no response text. If environment is not fetched, then should the execution be halted gracefully?

For 1, it wont happen but the loop has no control mechanism in place. Can this be written in better way? yes. 
This cause is a good fit for do-while loop. Otherwise in the case of while loop this can be written like this

var fetchEnvironmentPage = true;
while(fetchEnvironmentsPage)
some more code
fetchEnvironmentPage = apiResponse.HasNextPage
end of while
throw